test/mocks.py

```diff
@@ -0,0 +1,610 @@
+                       HIVE_VOICE, MODERATION_ROLES, VOICE, NITRO_BOOSTER, EVERYONE)
+from unittest.mock import AsyncMock, create_autospec, MagicMock
+from src.db.data_objects import BatteryType, Drone, DroneOrder, Storage
+from src.db.timer import Timer
+from src.drone_member import DroneMember
+from src.bot_utils import COMMAND_PREFIX
+from src.db.record import Record
+import src.channels as channels
+
+unique_id = 1
+
+emojis = [
+    'hex_a', 'hex_b', 'hex_c', 'hex_d', 'hex_e', 'hex_f', 'hex_g', 'hex_h', 'hex_i', 'hex_j', 'hex_k', 'hex_l',
+    'hex_m', 'hex_n', 'hex_o', 'hex_p', 'hex_q', 'hex_r', 'hex_s', 'hex_t', 'hex_u', 'hex_v', 'hex_w', 'hex_x',
+    'hex_y', 'hex_z', 'hex_1', 'hex_2', 'hex_3', 'hex_4', 'hex_5', 'hex_6', 'hex_7', 'hex_8', 'hex_9',
+    'blank', 'hex_slash', 'hex_dot', 'hex_questionmark', 'hex_exclamationmark', 'hex_comma', 'hex_dc',
+    BATTERY_FULL, BATTERY_MID, BATTERY_LOW, BATTERY_EMPTY, DRONE_EMOJI,
+]
+
+roles = [
+    INITIATE, ASSOCIATE, DRONE, STORED, DEVELOPMENT, ADMIN, MODERATION, HIVE_MXTRESS, SPEECH_OPTIMIZATION,
+    GLITCHED, ID_PREPENDING, IDENTITY_ENFORCEMENT, THIRD_PERSON_ENFORCEMENT, BATTERY_POWERED, BATTERY_DRAINED,
+    FREE_STORAGE, HIVE_VOICE, MODERATION_ROLES, VOICE, NITRO_BOOSTER, EVERYONE,
+]
+
+
+class TestBot(Bot):
+    context: Context = None
+
+
+class Mocks():
+    '''
+    Manage a mock guild.
+
+    This class creates a mock guild and provides factory methods for creating mock objects associated with the guild.
+    Creating a mock object will automatically add it to the guild.
+
+    Mocks are created with create_autospec() instead of Mock() so that accessing invalid properties or members will
+    cause an error, rather than creating a new mock property.
+    '''
+
+    _guild: MagicMock
+
+    _bot: Bot
+
+    def __init__(self):
+        self._guild = self.guild()
+
+        # Set up some default channels.
+        for name in ['general', 'hex-office', 'hexcorp-transmissions', 'hive-storage-chambers']:
+            self.channel(name)
+
+        # Set up default emoji.
+        for name in emojis:
+            self.emoji(name)
+
+        # Set up default roles.
+        for name in roles:
+            self.role(name)
+
+        self.hive_mxtress()
+
+        self._bot = self.bot()
+
+    def mock(*args, **kwargs) -> MagicMock:
+        return create_autospec(*args, **kwargs)
+
+    def get_guild(self) -> MagicMock:
+        '''
+        Get the mock Guild.
+        '''
+
+        return self._guild
+
+    def get_bot(self) -> Bot:
+        '''
+        Get the Bot, for testing bot commands.
+        '''
+
+        return self._bot
+
+    def get_cog(self) -> Cog:
+        '''
+        Get the first Cog on the bot.
+
+        The bot should usually only have one Cog, as added by the @cog() decorator.
+        '''
+
+        cogs = self.get_bot().cogs
+
+        if len(cogs) == 0:
+            raise Exception('Bot has no Cogs')
+
+        return cogs[next(iter(cogs))]
```

Comment:
I understand that this works, but is there a reason to use this instead of using `cogs[0]`? This may just be not knowing some arcane case where that would fail.

Reply:
No wait, why does this work? Doesn't `next()` already return the first element?